python/dgl/distributed/kvstore.py

```diff
@@ -0,0 +1,1008 @@
+    ip_config : str
+        Path of IP configuration file.
+    num_clients : int
+        Total number of KVClients that will be connected to the KVServer.
+    """
+    def __init__(self, server_id, ip_config, num_clients):
+        assert server_id >= 0, 'server_id (%d) cannot be a negative number.' % server_id
+        assert check_file_exists(ip_config), 'Cannot open file: %s' % ip_config
+        assert num_clients >= 0, 'num_clients (%d) cannot be a negative number.' % num_clients
+        # Register services on server
+        rpc.register_service(KVSTORE_PULL,
+                             PullRequest,
+                             PullResponse)
+        rpc.register_service(KVSTORE_PUSH,
+                             PushRequest,
+                             None)
+        rpc.register_service(INIT_DATA,
+                             InitDataRequest,
+                             InitDataResponse)
+        rpc.register_service(BARRIER, 
+                             BarrierRequest, 
+                             BarrierResponse)
+        rpc.register_service(REGISTER_PUSH, 
+                             RegisterPushHandlerRequest,
+                             RegisterPushHandlerResponse)
+        rpc.register_service(REGISTER_PULL,
+                             RegisterPullHandlerRequest,
+                             RegisterPullHandlerResponse)
+        rpc.register_service(GET_SHARED,
+                             GetSharedDataRequest,
+                             GetSharedDataResponse)
+        rpc.register_service(GET_PART_SHAPE,
+                             GetPartShapeRequest,
+                             GetPartShapeResponse)
+        rpc.register_service(SEND_META_TO_BACKUP,
+                             SendMetaToBackupRequest,
+                             SendMetaToBackupResponse)
+        # Store the tensor data with specified data name
+        self._data_store = {}
+        # Store the partition information with specified data name
+        self._part_policy = {}
+        # Basic information
+        self._server_id = server_id
+        self._server_namebook = rpc.read_ip_config(ip_config)
+        # TODO(chao): machine_id can be removed if we use new shared-memory API
+        self._machine_id = self._server_namebook[server_id][0]
+        self._group_count = self._server_namebook[server_id][3]
+        # We assume partition_id is equal to machine_id
+        self._part_id = self._machine_id
+        self._num_clients = num_clients
+        self._barrier_count = 0
+        # push and pull handler
+        self._push_handler = default_push_handler
+        self._pull_handler = default_pull_handler
+
+    @property
+    def server_id(self):
+        return self._server_id
+
+    @property
+    def barrier_count(self):
+        return self._barrier_count
+
+    def incr_barrier_count(self):
+        self._barrier_count +=1
+
+    def barrier_zero(self):
+        self._barrier_count = 0
```

Comment:
i feel it's not necessary to expose these APIs.

Reply:
Ok. I think we can just use a setter.